prov/sockets/src/sock_ep.c

```diff
@@ -119,6 +119,22 @@ static void sock_rx_ctx_close(struct sock_rx_ctx *rx_ctx)
 		sock_cntr_remove_rx_ctx(rx_ctx->comp.rem_write_cntr, rx_ctx);
 }
 
+static void sock_ep_release_ctx_rx_entries(struct sock_rx_ctx *rx_ctx) {
+	/* These are recv commands that have not completed before closing the RX_CTX.
+	   Release them all because some might be malloc'd. In principle the client
+	   should cancel all recvs before closing so there should be none left
+	   here. But regardless of that correctness, ep should not leak memory. */
+	struct dlist_entry *entry, *tmp;
+	struct sock_rx_entry *rx_entry;
+	fastlock_acquire(&rx_ctx->lock);
+	dlist_foreach_safe(&rx_ctx->rx_entry_list, entry, tmp) {
+		rx_entry = container_of(entry, struct sock_rx_entry, entry);
+		dlist_remove(&rx_entry->entry);
+		sock_rx_release_entry(rx_entry);
+	}
+	fastlock_release(&rx_ctx->lock);
```

Comment:
I may not be understanding the change here.  Based on the comment, the lock isn't necessary; we're about to free the rx_ctx.  Also, why not call this from only sock_rx_ctx_free(), rather than from several places?